fc-community-contracts/src/FactChainSFT.sol

```diff
@@ -0,0 +1,80 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.20;
+
+import {ERC1155URIStorage} from "openzeppelin-contracts/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol";
+import {ERC1155} from "openzeppelin-contracts/contracts/token/ERC1155/ERC1155.sol";
+import {Ownable} from "./utils/Ownable.sol";
+
+import {Arrays} from "openzeppelin-contracts/contracts/utils/Arrays.sol";
+
+interface IFactChainSFTEvents {
+    event FactchainNFTContractUpdated(address factchainNFTContract);
+    event FactchainBuildersRewarded(uint256 amount);
+    event CreatorRewarded(address creator, uint256 amount);
+}
+
+interface IFactChainSFT is IFactChainSFTEvents {
+    // Errors
+    error SupplyExhausted();
+    error ValueError();
+    error BadMintPrice();
+    error NegativeBalance();
+    error FailedToReward();
+    error ReservedToFactChain();
+}
+
+contract FactChainSFT is Ownable, ERC1155URIStorage, IFactChainSFT {
+    address public FACTCHAIN_NFT_CONTRACT;
+    uint256 public constant FACTCHAINERS_MINT_SUPPLY = 42;
+    uint256 public constant MINT_PRICE = 1_000_000;
+
+    mapping(uint256 id => uint256) private _supply;
+
+    using Arrays for address[];
+
+    constructor(address _owner)
+        Ownable(_owner)
+        ERC1155("https://gateway.pinata.cloud/ipfs/")
+    {
+        _setBaseURI("https://gateway.pinata.cloud/ipfs/");
+    }
+
+    function setFactchainNFTContract(address _factchainNFTContract) public onlyOwner {
+        FACTCHAIN_NFT_CONTRACT = _factchainNFTContract;
+        emit FactchainNFTContractUpdated(_factchainNFTContract);
+    }
+
+    function mint(address[] memory raters, string memory ipfsHash, uint256 id) public returns (uint256) {
```

Comment:
could be renamed for clarity

Reply:
- add creator to this function, and store it in contract alongside id
- or have this contract call `FACTCHAIN_NFT_CONTRACT` to know creator address?